lm_eval/tasks/robustness/README.md

```diff
@@ -0,0 +1,76 @@
+# ROBUSTNESS
+
+## Paper
+Title: [PAPER TITLE HERE] <!--(PAPER_LINK_HERE) -->
+
+[ABSTRACT PLACEHOLDER]
+
+
+## Citation
+```bib
+[Citation placeholder]
+```
+
+## Groups
+
+- `robustness_mmlu_pro`: three 0-shot robutstness tasks on MMLU-PRO dataset
+
+- `robustness_agieval`: three 0-shot robutstness tasks on the AGIEVAL datasets multiple choice questions subsets:  `'agieval-sat-math'`, `'agieval-lsat-lr'`, `'agieval-lsat-rc'`, `'agieval-logiqa-en'`, `'agieval-aqua-rat'`, `'agieval-sat-en'`, `'agieval-lsat-ar'` 
+
+- `robustness_mmlu_pro_fewshot`: a 5-shot robutstness task on MMLU-PRO dataset
+
+## Tasks
+
+Both `robustness_mmlu_pro` and `robustness_agieval` contain the following 3 tasks:
+
+* Option format robustness: `option_format_robustness_mmlu_pro`, `option_format_robustness_agieval`
+
+* Option order robustness: 
+`option_order_robustness_mmlu_pro`, `option_order_robustness_agieval`
+
+* Prompt robustness: 
+`prompt_robustness_mmlu_pro`, 
+`prompt_robustness_agieval`
+
+### Option format robustness
+
+Measures the model's robustness towards the following option formats:
+<br>
+- Uppercase latter + colon (`A: option_1` `B: option_2` ... )
+- Uppercase latter + parenthesis (`A) option_1` `B) option_2` ... )
+- Uppercase latter + dot (`A. option_1` `B. option_2` ... )
+- Numeral + dot (`1. option_1` `2. option_2` ... )
+- Lowercase latter + colon (`a: option_1` `b: option_2` ... )
+- Roman numeral + colon (`I: option_1` `II: option_2` ... )
+
+### Option order robustness
+
+Measures the model's robustness towards the placment of the correct answer in the options list by swapping the correct answer with all the other possible options.
+
+### Prompt robustness
+
+Measures the model's robustness towards 10 different prompts. list of the prompts can be found in the `./prompt_templates.json` file under the key `prompt_robustness`.
+
+### Fewshot Prompt robustness
+
+Measures the model's robustness towards the same 10 prompts as the former task in a fewshot manner. This task is only available for MMLU-PRO.
+
+For evaluating `robustness_mmlu_pro_fewshot` it is required to pass the `"--fewshot_as_multiturn"` flag as well as set the `"--num_fewshot 5"`,
```

Comment:
is there a way to set default for the overall robustness task?

Reply:
No actually in their current implementation `--fewshot_as_multiturn` parameter could only be passed by argparser, there is no way to add this into yaml without changing some of the logic in the their evaluator class.
What concerns `--num_fewshot`, even though we cam pass it through yaml, but they have a assertion check where enforce passing `num_fewshot`>0 when `fewshot_as_multiturn` is True, but in this assertion they are looking to the parameter passed through argparser not the one in the yaml.

TLDR: there is not a way for setting this default for the overall robustness task without changes in their evaluator class.